### PR TITLE
ART-8042 Add rpm support in golang report

### PR DESCRIPTION
**Pull request by thegreyd**
Follow up to https://github.com/openshift-eng/art-tools/pull/337

https://issues.redhat.com/browse/ART-8042

```
$ ./elliott --quiet go:report --ocp-versions 4.14,4.11,4.12,4.13,4.15,4.16 --ignore-rhel

4.14: [{'go_version': '1.20.12', 'building_image_count': 219, 'building_rpm_count': 2}, {'go_version': '1.19.13', 'building_image_count': 2}]
4.11: [{'go_version': '1.18.10', 'building_image_count': 194, 'building_rpm_count': 2}, {'go_version': '1.19.13', 'building_image_count': 3}]
4.12: [{'go_version': '1.19.13', 'building_image_count': 207, 'building_rpm_count': 2}]
4.13: [{'go_version': '1.19.13', 'building_image_count': 215, 'building_rpm_count': 2}]
4.15: [{'go_version': '1.20.12', 'building_image_count': 223, 'building_rpm_count': 2}, {'go_version': '1.19.13', 'building_image_count': 1}]
4.16: [{'go_version': '1.21.3', 'building_image_count': 184, 'building_rpm_count': 2}, {'go_version': '1.20.12', 'building_image_count': 40}, {'go_version': '1.19.13', 'building_image_count': 1}]
```

### CODE REVIEW

File: elliott/elliottlib/cli/get_golang_report_cli.py [hunk 1]
```diff
@@ -74,18 +71,69 @@ def get_golang_report_cli(runtime, ocp_versions, ignore_rhel, output):
...
+        out = sorted(out, key=lambda x: x['building_image_count'] if 'building_image_count' in x
+                     else x['building_rpm_count'], reverse=True)
         results[ocp_version] = out
 
     if output == 'json':
         print(json.dumps(results, indent=4))
     else:
         for ocp_version, result in results.items():
             green_print(f'{ocp_version}: {result}')
+
```
Comment (locriandev) [inline elliott/elliottlib/cli/get_golang_report_cli.py:118]: Needs an extra blank line

File: elliott/elliottlib/cli/get_golang_report_cli.py [hunk 2]
```diff
@@ -74,18 +71,69 @@ def get_golang_report_cli(runtime, ocp_versions, ignore_rhel, output):
                 v = golang_streams[b]
                 golang_streams_images[v].append(image_name)
 
+        # Analyze defined rpms
```
Comment (locriandev) [inline elliott/elliottlib/cli/get_golang_report_cli.py:74]: This function has grown too big, can we split it up?

**Review by locriandev — COMMENTED** [review]
Overall LGTM. Noticing how the execution can take quite a while, maybe we can use threads to speed it up a bit? e.g. the nested loops inside the main one could be run in separate functions and use `exectools.parallel_exec()`

File: elliott/elliottlib/cli/get_golang_report_cli.py (hunk 2 above)
Comment (thegreyd) [reply elliott/elliottlib/cli/get_golang_report_cli.py:74]: Don't want to spend too much time on this, if it's okay would like to merge this change in and think of refactor down the line.

**Pull request merged**